corehq/apps/users/tasks.py

```diff
@@ -312,16 +315,24 @@ def process_reporting_metadata_staging():
     )[:100]
     with transaction.atomic():
         for record in records:
+            user = CouchUser.get_by_user_id(record.user_id, record.domain)
+            if not user or user.is_deleted():
+                continue
+
+            save = False
             if record.received_on:
-                mark_latest_submission(
-                    record.domain, record.user_id, record.app_id, record.build_id,
-                    record.xform_version, record.form_meta, record.received_on
+                save = mark_latest_submission(
+                    record.domain, user, record.app_id, record.build_id,
+                    record.xform_version, record.form_meta, record.received_on, save=False
```

Comment:
nit: i think the naming of the two `saves` is a bit confusing.
the one passed as argument is telling if the user should be saved or not, if needed.
the return value is to save the user or not.
having different names might be helpful
`save_user` for returned and `skip_user_save` for the argument (after reverting the default value)

Reply:
Nice. I like the idea of having `save_user`. I used `save` as the argument in the function because it stays consistent with our other methods (like `update_device_meta`). Though I'm not opposed to changing those method's arguments. Will do a followup PR